File: number-of-zero-filled-subarrays.ts
{
    function zeroFilledSubarray(nums: number[]): number {
        let r = 0,
            c = 0;
    
        for (const num of nums) {
            if (num == 0)
                c += 1;
            else 
                c = 0;

            r += c;
        }

        return r;
    };

    console.log(zeroFilledSubarray([0,0,0,2,0,0]))
}

File: number-of-zero-filled-subarrays.rs
pub fn zero_filled_subarray(nums: Vec<i32>) -> i64 {
    let mut r = 0;
    let mut c = 0;

    for num in nums {
        if num == 0 {
            c += 1;
        } else {
            c = 0;
        }

        r += c;
    }

    return r;
}
